compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -1152,7 +1164,7 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
             self.err_handler(),
             generics.params.iter().map(|param| {
                 let ident = Some(param.ident.to_string());
-                let (kind, ident) = match &param.kind {
+                let (ord_kind, ident) = match &param.kind {
```

Comment:
Since we pass the `GenericParamKind` now, we can compute `ord_kind` within `validate_generic_param_order`, rather than doing it here.

Reply:
I moved all of the logic done here into `validate_generic_param_order`, was there a reason we passed an iterator of a bunch of pieces of the params instead of an iterator of the `GenericParam`s themselves?